lib/Sema/CSDiag.cpp

```diff
@@ -7525,7 +7525,10 @@ bool FailureDiagnosis::diagnoseMemberFailures(
 
   // If this is a tuple, then the index needs to be valid.
   if (auto tuple = baseObjTy->getAs<TupleType>()) {
-    StringRef nameStr = memberName.getBaseIdentifier().str();
+    if (memberName.isSpecial())
+      return false;
+
+    StringRef nameStr = memberName.getBaseName().userFacingName();
```

Comment:
Since it gets the user facing name here and not the identifier, do you need the check above?

Reply:
Yeah, i think I can redo this whole check actually to make it better.

Reply:
But to answer original question - yes, it's necessary because later on `getNamedElementId` needs to use identifier and not just a string...

Reply:
I made it so we assume that member is an index into the tuple only if it's not special, otherwise we diagnose directly, I think that is more straightforward way to think about this.